packages/backend/src/api/event/eventRouter.ts

```diff
@@ -0,0 +1,176 @@
+import { OpenAPIRegistry } from '@asteasolutions/zod-to-openapi';
+import express, { Request, Response, Router } from 'express';
+import { StatusCodes } from 'http-status-codes';
+import { ResponseStatus, ServiceResponse } from 'src/common/models/serviceResponse';
+import { z, ZodError } from 'zod';
+
+import { createApiResponse } from '../../api-docs/openAPIResponseBuilders';
+import { handleServiceResponse, validateRequest } from '../../common/utils/httpHandlers';
+import { EventSchema, RegistrationSchema, SaveEventSchema } from '../event/eventModel';
+import { eventService } from '../event/eventService';
+import {
+  CreateEventSchema,
+  GetAllEventSchema,
+  GetEventRequest,
+  GetEventSchema,
+  GetRegistrationSchema,
+  RegisterEventSchema,
+  SaveEventRequestSchema,
+  UpdateEventSchema,
+} from './eventRequest';
+
+export const eventRegistry = new OpenAPIRegistry();
+
+eventRegistry.register('Event', EventSchema);
+eventRegistry.register('UserEventRegistration', RegistrationSchema);
+
+export const eventRouter: Router = (() => {
+  const router = express.Router();
+
+  eventRegistry.registerPath({
+    method: 'get',
+    path: '/events',
+    tags: ['Event'],
+    responses: createApiResponse(z.array(EventSchema), 'Success'),
+  });
+
+  router.get('/', validateRequest(GetAllEventSchema), async (req: Request, res: Response) => {
+    const { pageIndex, pageSize, typeId, title } = req.query as unknown as GetEventRequest;
+    const serviceResponse = await eventService.findAll({
+      pageIndex,
+      pageSize,
+      typeId,
+      title,
+    });
+    handleServiceResponse(serviceResponse, res);
+  });
+
+  eventRegistry.registerPath({
+    method: 'get',
+    path: '/events/{id}',
+    tags: ['Event'],
+    responses: createApiResponse(z.array(EventSchema), 'Success'),
+  });
+
+  router.get('/:id', validateRequest(GetEventSchema), async (req: Request, res: Response) => {
+    console.log('req.params', req.params);
+    const id = req.params.id as string;
+    const serviceResponse = await eventService.findById(id);
+    handleServiceResponse(serviceResponse, res);
+  });
+
+  eventRegistry.registerPath({
+    method: 'post',
+    path: '/events',
+    tags: ['Event'],
+    request: {
+      params: CreateEventSchema,
+    },
+    responses: createApiResponse(EventSchema, 'Success'),
+  });
+
+  router.post('/', async (req: Request, res: Response) => {
+    try {
+      const parsedEvent = CreateEventSchema.parse({ body: req.body });
```

Comment:
You can do `const createEventRequest = req.body as unknown as CreateEventRequest; ` here like the jobs instead of doing parse. Check the job router implementation and the validate() middleware. Also this would allow you to not check the event.body in the service.